### ISSUE TITLE
mean with squeeze=False works incorrectly under torch

### ISSUE THREAD

**Issue by vabor112** [wesselb/lab#21]
If `axis` is None (or not provided), `mean(torch_tensor, squeeze=squeeze)` will ignore the value of `squeeze`.

This is obvious from the [implementation](https://github.com/wesselb/lab/blob/1e06d702d01fbb8318c709968f612438159fd8e7/lab/torch/generic.py#L293):

```python
@dispatch
def mean(a: Numeric, axis: Union[Int, None] = None, squeeze: bool = True):
    if axis is None:
        return torch.mean(a)
    else:
        return torch.mean(a, dim=axis, keepdim=not squeeze)
```

Why not just change the implementation to the following? It works as expected for me and handles axis=`None` okay.
(Disclaimer: I tested it only under torch 2.2.0)

```python
@dispatch
def mean(a: Numeric, axis: Union[Int, None] = None, squeeze: bool = True):
        return torch.mean(a, dim=axis, keepdim=not squeeze)
```

**Comment by wesselb** [comment]
@vabor112 Hmm, I don't recall exactly why I implement this this way. Perhaps `torch` was less flexible at the time, or I simply didn't think things through.

Out of interest, is it not correct to ignore the value of `squeeze` whenever `axis` is not provided? Or, if `x = np.ones((5, 5))`, would you expect `B.mean(x, squeeze=False)` to give a `(1, 1)` tensor? That's an interesting use case that I haven't considered.

**Comment by wesselb** [comment]
Interestingly, only `torch.mean` seems to accept `dim=None`. I've simplified the implementation according to your suggestion. Thanks! :)

**Comment by vabor112** [comment]
> Or, if x = np.ones((5, 5)), would you expect B.mean(x, squeeze=False) to give a (1, 1) tensor? That's an interesting use case that I haven't considered.

I think this is what I needed, yes. But I'm not 100% sure since it was a pretty long time ago. Thanks for fixing it!
